services/brig/test/integration/API/Provider.hs

```diff
@@ -510,9 +513,35 @@ testAddRemoveBotTeam config crt db brig galley cannon = withTestService config c
     addBot brig uid1 pid sid cidFail !!! do
         const 403 === statusCode
         const (Just "invalid-conversation") === fmap Error.label . decodeBody
-
     testAddRemoveBotUtil pid sid cid u1 u2 h sref buf brig galley cannon
 
+testBotTeamOnlyConv :: Maybe Config -> FilePath -> DB.ClientState -> Brig -> Galley -> Cannon -> Http ()
+testBotTeamOnlyConv config crt db brig galley cannon = withTestService config crt db brig defServiceApp $ \sref buf -> do
+    (u1, u2, _h, _tid, cid, pid, sid) <- prepareBotUsersTeam brig galley sref
+    let (uid1, uid2) = (userId u1, userId u2)
+    -- Make the conversation team-only and check that the bot can't be added
+    -- to the conversation
+    setAccessRole uid1 cid TeamAccessRole
+    addBot brig uid1 pid sid cid !!! do
+        const 403 === statusCode
+        const (Just "invalid-conversation") === fmap Error.label . decodeBody
+    -- Make the conversation allowed for guests and add the bot successfully
+    setAccessRole uid1 cid NonActivatedAccessRole
+    bid <- addBotConv brig cannon uid1 uid2 cid pid sid buf
+    let buid = botUserId bid
+    -- Make the conversation team-only again and check that the bot has been removed
+    setAccessRole uid1 cid TeamAccessRole
+    WS.bracketR cannon uid1 $ \ws -> do
+        _ <- waitFor (5 # Second) not (isMember galley buid cid)
+        getBotConv galley bid cid !!!
+            const 404 === statusCode
+        svcAssertMemberLeave buf buid [buid] cid
+        wsAssertMemberLeave ws cid buid [buid]
```

Comment:
This one fails: `Exception: Timeout: No matching notification received`. I've no idea why :(